instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActiveRecord
+      # The Instrumentation class contains logic to detect and install the ActiveRecord instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('5.2.0')
+
+        install do |_config|
+          require_dependencies
+          patch
+        end
+
+        present do
+          defined?(::ActiveRecord)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        private
+
+        def insert_class_methods_supported?
+          gem_version >= Gem::Version.new('6.0.0')
+        end
+
+        def gem_version
+          Gem.loaded_specs['activerecord'].version
+        end
+
+        def patch
+          # ::ActiveRecord::Querying.prepend(Patches::Querying)
```

Comment:
Curious about the comments in this method - are these documenting what's actually being patched? Or are they just left over from earlier efforts?

Reply:
I'll clean this up and leave a proper comment about what's being patched.  Effectively what I was doing before was patching the modules that were being included/extended in ActiveRecord::Base.  Now I'm patching ActiveRecord::Base directly, it appears to consistently have the desired behaviour.